Style/Frontend/index.md

```diff
@@ -0,0 +1,1422 @@
+  console.log(user2.name === 'Heewon Ahn') // true
+
+  // similar for arrays, use spread syntax to copy
+  const array = [1, 2, 3, 4];
+  const arrayCopy = [...array];
+  ```
+
+- Use functional components instead of class components
+
+  > There is one singular difference between React class components and functional components - class components mutate themselves, functional components are immutable. A class component will simply mutate it's own internal references when props change where as an entirely new functional component will be created if it's props change. You can imagine the completed application as a series of stateful snapshots over time that can each be traced back to the exact set of props that were given to it.
+
+#### A note about Immer
+
+So now this is going to sound like it completely contradicts the above statements, but there is one specific scenario where writing mutating code is actually fine, and that is in redux toolkit reducers. Redux toolkit makes use of the [Immer](https://immerjs.github.io/immer/) library, which means that redux reducers receive a *copy* of the current state called a draft which can be modified with the usual methods, then that draft is used to create the new state.
+
+```jsx
+.addCase(createAlpacaACHRelationship.fulfilled, (state, action) => {
+  state.achList.push(action.payload.relationship);
+})
+```
+
+This is a real redux reducer we use, as you can see, we directly manipulate the state by pushing to it. Normally this is not a good idea, but Immer is working behind the scenes to make this possible. As you can see, we don't need to spread the state or do any redux weirdness, Immer just handles it for us, and the code is much cleaner, compact, and easier to understand.
+
+## Naming Conventions
+
+- Global constants must be `SCREAMING_SNAKE_CASE`
+
+  ```jsx
+  // bad
+  const globalConst = 5;
+
+  // good
+  const GLOBAL_CONST = 5;
+  ```
+
+- React components must be in `PascalCase` (Each word capitalized)
+
+  ```jsx
+  // bad
+  const reactComponent = () => <View />;
+
+  // good 
+  const ReactComponent = () => <View />;
+  ```
+
+- Functions and hooks regardless of their location must be `camelCase`
+
+  ```jsx
+  // bad
+  const myfunc = (x) => x**2;
+  const MyFunc = (x) => x**2;
+
+  // good
+  const myFunc = (x) => x**2;
+
+  const Component = ({}) => {
+    // bad
+    const myfunc = (x) => x**2;
+    const MyFunc = (x) => x**2;
+
+    // good
+    const myFunc = (x) => x**2;
+  }
+  ```
+
+- Local constants and testIDs must be `camelCase`
+
+  ```jsx
+  const Component = ({}) => {
+    // bad
+    const LocalConst = 5;
+    const localobj = {a: 1};
+    
+    // good
+    const localConst = 5;
+    const localObj = {a: 1};
+    
+    // bad
+    const SubComponent = () => <View testID="SubComponent.View" />;
+    	
+    // good
+    const SubComponent = () => <View testID="subComponent.view" />;
+  }
+  ```
+
+- No leading or trailing underscores
```

Comment:
This rule I think has some contradictions in our code but despite that I think it does have some value, would only make sense to keep if we want to commit to it though with the `no-underscore-dangle` plugin